### ISSUE TITLE
materials generated from .gltf file are too big in filesize

### ISSUE THREAD

**Issue by ruki3** [godotengine/godot#34641]
**Godot version:**
3.1.2
**OS/device including version:**
Windows 10
**Issue description:**
When you import fully textured .gltf scenes Godot automatically generates .material files for you. Unfortunately there's a flaw in it, because, for example - a door material is 13 mb in size while all the textures are barely 300kb in size. Also, the exported gltf is only 2,6 mb in size. Something's totally not right here.
**Steps to reproduce:**
Make a model in blender, make materials, export to gltf and finally export to godot.
**Minimal reproduction project:**
[material-bug.zip](https://github.com/godotengine/godot/files/4006125/material-bug.zip)

here's a screenshot of all of the wooden floor material files - they're only 718 kb in file size, while the material is 9,2 mb (!)
![Screenshot_6](https://user-images.githubusercontent.com/57419448/71530481-3d0de300-28ea-11ea-88ba-5d6c51d242b5.png)

**Comment by Calinou** [comment]
Maybe `.material` files are embedding textures in uncompressed form.

**Comment by ruki3** [comment]
Is there any way to avoid that without running a custom build of Godot? I've tried messing with the model's import settings, but couldn't find anything related to texture compression.

**Comment by Calinou** [comment]
cc @fire, as he has some knowledge about the glTF 2.0 importer.

**Comment by fire** [comment]
Looking!

```
    "buffers" : [
        {
            "byteLength" : 1991968,
            "uri" : "data:application/octet-stream;base64,
```

You embedded something binary in the gltf2 file. 

I would try exporting as scn to check file size. There was other comments about text formats for example base64'd png files encoded as base64 versions.

**Comment by Calinou** [comment]
Indeed, text formats like `.tscn` will encode binary data as Base64, which is at least 33% larger than the raw binary data.

**Comment by ruki3** [comment]
I didn't embed anything in the gltf2 file, at least not on purpose. Could it be that some addons are causing the problem? I also tried disabling some functions that I don't need at the export stage (animations, skinning etc.) but it didn't reduce the size of the file. Thanks for the suggestion about the scn export, it doesn't include the textures inside the file like gltf, but at least I get to keep the size of my project small, as .dae didn't work for me aswell (I swear Godot hates me)

**Comment by aaronfranke** [comment]
Any update? I have a `.material` file which is 41.9 MB, but the image it contains is only 170.3 kB when saved as a `.png` file. This is ridiculously huge.

**Comment by KoBeWi** [comment]
Might be duplicate of #31632

**Comment by Bleuzen** [comment]
Is there any update on this now? Would love to use gltf, but with this bug it is unusable.
As a workaround for now we can use .dae or gltf separate formats.

Here are my tests so far in different formats (all exported from Blender and imported to Godot).
I*m using this as example: https://ln.sync.com/dl/41dc1c2b0/wxfm2fzf-w2ggnk53-i7eyv6zj-caq828di (from video https://www.youtube.com/watch?v=XMfiojwtuKo)

**.DAE**
total size: 213 kb
![grafik](https://user-images.githubusercontent.com/12885163/94371700-b8f6bf00-00f8-11eb-9b41-627f76252dba.png)

**GLTF SEPARATE**
total size: 98 kb
![grafik](https://user-images.githubusercontent.com/12885163/94371714-cf047f80-00f8-11eb-9fdf-c16f32a26b9d.png)

GLTF binary (and embedded):
total size: 3.1 mb
![grafik](https://user-images.githubusercontent.com/12885163/94371728-f3f8f280-00f8-11eb-9aa4-9c5ac835ab74.png)

**Comment by Calinou** [comment]
@Bleuzen As far as I know, nobody found a solution to this yet.

**Comment by fire** [comment]
For me using gltf embedded where you export  text gltf and the textures to a textures folder as png etc works the best.

**Comment by abrasivetroop** [comment]
Any updates on this? My .material file sizes were too big. I realized it was the textures that were too big in size and I had to go through every file and change the texture compression mode to lossless compression. By default textures imported from .gltf material files are uncompressed. If textures were lossless compressed by default this wouldn't be a huge problem. 

![image](https://user-images.githubusercontent.com/47716519/209001581-369b20fd-a4d4-4ec3-9486-f7cc133a5f5e.png)

Doing this reduced file sizes by 10 Mbs which also reduced scene loading times. My project size reduced by 1GB too after doing this for every material file in my project. There is no way of changing this in Import Defaults too. Sorry if I am missing something.

**Comment by KoBeWi** [comment]
> There is no way of changing this in Import Defaults too. Sorry if I am missing something.

Profile -> Set as Default in Import dock.

**Comment by abrasivetroop** [comment]
@KoBeWi I am sorry can't find the setting you are talking about

**Comment by abrasivetroop** [comment]
![image](https://user-images.githubusercontent.com/47716519/209005205-53228146-f642-4568-9527-fd2910fd1ecc.png)
@KoBeWi  Are you talking about this? If you can't set a import default for materials that are imported with .gltf files from the import menu.

**Comment by KoBeWi** [comment]
You mean these textures ignore defaults? Do the defaults work correctly for regular textures?

**Comment by abrasivetroop** [comment]
Yes .gltf material files import texture with no compression and they ignore the import defaults for textures. I had to manually set the textures for each material to lossless compression and I will need to do this every time I import a gltf model with embedded textures. Unless there is a way to set an import default for this :/

**Comment by fire** [comment]
Work for this was being done in https://github.com/godotengine/godot/pull/62499

It's blocked on astc support being merged

**Comment by abrasivetroop** [comment]
@fire Awesome! When do you think this fix will be released?

**Comment by Calinou** [comment]
> When do you think this fix will be released?

There is no ETA, but note that ASTC support will only be added in Godot 4 – it won't be backported to 3.x, as there's no Basus Universal support there.

**Comment by abrasivetroop** [comment]
> There is no ETA, but note that ASTC support will only be added in Godot 4 – it won't be backported to 3.x.

I see, Godot setting built-in gltf textures to lossless compression would be enough for now. It is just a hassle manually setting all of them :/ 

And thanks for everyone informing me about this!

**Comment by lyuma** [comment]
This issue was solved in time for the release of 4.0, by extracting textures from glb files. It still exists in 3.x
